kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/DroolsAuthoringPerspective.java

```diff
@@ -1,152 +1,166 @@
-import org.kie.workbench.common.services.shared.preferences.ApplicationPreferences;
-import org.kie.workbench.common.widgets.client.handlers.NewResourcePresenter;
+import com.google.gwt.user.client.Window;
+import org.jboss.errai.common.client.api.Caller;
+import org.jboss.errai.common.client.api.RemoteCallback;
 import org.kie.workbench.common.widgets.client.handlers.NewResourcesMenu;
-import org.kie.workbench.common.widgets.client.menu.RepositoryMenu;
 import org.kie.workbench.common.workbench.client.PerspectiveIds;
 import org.kie.workbench.common.workbench.client.docks.AuthoringWorkbenchDocks;
 import org.kie.workbench.drools.client.resources.i18n.AppConstants;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.backend.vfs.VFSService;
 import org.uberfire.client.annotations.Perspective;
 import org.uberfire.client.annotations.WorkbenchMenu;
 import org.uberfire.client.annotations.WorkbenchPerspective;
 import org.uberfire.client.mvp.PlaceManager;
+import org.uberfire.client.workbench.PanelManager;
 import org.uberfire.client.workbench.panels.impl.MultiListWorkbenchPanelPresenter;
-import org.uberfire.mvp.Command;
+import org.uberfire.lifecycle.OnOpen;
+import org.uberfire.mvp.PlaceRequest;
 import org.uberfire.mvp.impl.DefaultPlaceRequest;
+import org.uberfire.workbench.model.PanelDefinition;
+import org.uberfire.workbench.model.PartDefinition;
 import org.uberfire.workbench.model.PerspectiveDefinition;
 import org.uberfire.workbench.model.impl.PerspectiveDefinitionImpl;
 import org.uberfire.workbench.model.menu.MenuFactory;
-import org.uberfire.workbench.model.menu.MenuPosition;
 import org.uberfire.workbench.model.menu.Menus;
 
-/**
- * A Perspective for Rule authors
- */
 @ApplicationScoped
 @WorkbenchPerspective(identifier = PerspectiveIds.AUTHORING, isTransient = false)
 public class DroolsAuthoringPerspective {
 
     private AppConstants constants = AppConstants.INSTANCE;
 
-    @Inject
-    private NewResourcePresenter newResourcePresenter;
-
     @Inject
     private NewResourcesMenu newResourcesMenu;
 
     @Inject
     private PlaceManager placeManager;
 
     @Inject
-    private RepositoryMenu repositoryMenu;
+    private PanelManager panelManager;
 
     @Inject
-    private AuthoringWorkbenchDocks docks;
+    private Caller<VFSService> vfsServices;
 
     @Inject
-    private ExamplesWizard wizard;
+    private AuthoringWorkbenchDocks docks;
+
+    private String explorerMode;
+    private String projectPathString;
+    private boolean projectEditorDisableBuild;
+
+    private final List<PlaceRequest> placesToClose = new ArrayList<PlaceRequest>();
 
     @PostConstruct
-    public void setup() {
-        docks.setup( PerspectiveIds.AUTHORING, new DefaultPlaceRequest( "org.kie.guvnor.explorer" ) );
+    public void init() {
+        explorerMode = ( ( Window.Location.getParameterMap().containsKey( "explorer_mode" ) ) ? Window.Location.getParameterMap().get( "explorer_mode" ).get( 0 ) : "" ).trim();
```

Comment:
@Rikkola would you mind reformatting this PR, please, so we have consistent code style?